src/playground/sample_data_ingestion/sample_data_plugins/sample_data_plugin.py

```diff
@@ -0,0 +1,145 @@
+from datetime import datetime, timedelta
+from os import environ, path, remove
+from json import dumps, load, loads
+from shutil import copyfileobj
+from requests import request
+from base64 import b64encode
+from time import mktime
+import gzip
+
+
+"""
+Generic Plugin class: initialization is defined here
+Arguments:
+    - index_name: The name of the source index
+    - payload: The JSON body of which to call create API requests
+    - base_url: The IP address/url where the OpenSearch backend lies
+"""
+class Plugin:
```

Comment:
are you planing to add all the plugins here? or create a generic plugin call, then separate file for each plugins ?

Reply:
+1, you could have a base file for plugin class, and then have separate files for different plugins

Reply:
Done, all of the classes are on separate files